### PR TITLE
Lesson 20 Wiki Notes

### PR DESCRIPTION
**Pull request by gildedgardenia**

### CODE REVIEW

File: Class-19-(Authentication).md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# Class 19: Authentication
+
+## In the prep work this week, the students learned
+
+1. What authentication is.
+1. What sessions and cookies are and how sessions and cookies are used to verify a user's identity.
+1. The importance of encryption.
+1. Why developers use one-way encryption to protect users' passwords.
+1. How to use Spring Security in an application to enable authentication.
```
Comment (chrisbaylewis) [inline Class-19-(Authentication).md]: We aren't actually using Spring Security in this lesson. We should say something like "How to use request filtering in an application..."

File: Class-19-(Authentication).md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+1. Students are almost done with the class! One class left!
+1. Assignment #4 is due next class! 
+1. Check with your course manager for any additional announcements.
+
+## Large Group Time (Instructor)
+
+### Class 19 Topics That Require Careful Attention
+1. Review hashing and one-way encryption. Remind students that passwords are not stored, but the encrypted password is stored.
+1.
```
Comment (chrisbaylewis) [inline Class-19-(Authentication).md]: Was there another item here? Or just a stray list number?

File: Class-19-(Authentication).md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+## Large Group Time (Instructor)
+
+### Class 19 Topics That Require Careful Attention
+1. Review hashing and one-way encryption. Remind students that passwords are not stored, but the encrypted password is stored.
+1. 
+
+## Small Group Time: Class 19 Studio (TA Notes)
+
+1. The studio is focused on adding authentication to ``coding-events``.
```
Comment (chrisbaylewis) [inline Class-19-(Authentication).md]: We don't have a studio at this point. We talked about potentially having them add authentication to TechJobs, but unless they have finished the assignment that won't be a great idea. We should revisit this as a group and decide what to do.

File: Class-20-(REST).md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+## Large Group Time (Instructor)
+
+### Class 20 Topics That Require Careful Attention
+1. Review the characteristics of REST. REST is deceptively simple, but so important for the students to understand!
+1. Go over the Adopt-a-Pet examples! Highlight the URL portions of the URLs for different pet searches.
+1. Make sure to leave time to do a temperature check for the previous lesson. Did everyone finish the studio? Is everyone getting the concepts? Class 19 had a lot of material in it.
+
+## Small Group Time: Class 20 Studio (TA Notes)
+
+1. Make sure to get every student through the Spring Security studio! By the end of the class, everyone should have a fully-functioning web application with authentication!
```
Comment (chrisbaylewis) [inline Class-20-(REST).md]: Similar to above...need to decide what goes in the studio.

**Review by chrisbay — CHANGES_REQUESTED** [review]
Looks good overall, though we'll need to decide what to do about the studio for lesson 19 before this can be merged.

File: Class-19-(Authentication).md (hunk 1 above)
Comment (gildedgardenia) [reply Class-19-(Authentication).md]: Fixed!

File: Class-19-(Authentication).md (hunk 2 above)
Comment (gildedgardenia) [reply Class-19-(Authentication).md]: Added some more!

File: Class-19-(Authentication).md (hunk 3 above)
Comment (gildedgardenia) [reply Class-19-(Authentication).md]: Added some tips and tricks!

File: Class-20-(REST).md (hunk 4 above)
Comment (gildedgardenia) [reply Class-20-(REST).md]: We didn't have a studio planned for this class so I thought it was going to be a catch-up day.

**Review by chrisbay — APPROVED** [review]

**Pull request merged**